modules/imgproc/src/resize.cpp

```diff
@@ -50,12 +50,1326 @@
 #include "precomp.hpp"
 #include "opencl_kernels_imgproc.hpp"
 #include "hal_replacement.hpp"
+#include "opencv2/core/hal/intrin.hpp"
 
 #include "opencv2/core/openvx/ovx_defs.hpp"
 #include "resize.hpp"
 
+#include "opencv2/core/softfloat.hpp"
+
 using namespace cv;
 
+#ifndef CV_ALWAYS_INLINE
+    #ifdef CV_ALWAYSINLINE
+        #define CV_ALWAYS_INLINE CV_ALWAYSINLINE
+    #else
+        #if defined(__GNUC__) && (__GNUC__ > 3 ||(__GNUC__ == 3 && __GNUC_MINOR__ >= 1))
+            #define CV_ALWAYS_INLINE inline __attribute__((always_inline))
+        #elif defined(_MSC_VER)
+            #define CV_ALWAYS_INLINE __forceinline
+        #else
+            #define CV_ALWAYS_INLINE inline
+        #endif
+    #endif
+#endif
+
+namespace
+{
+
+class fixedpoint64
+{
+private:
+    static const int fixedShift = 32;
+    static const int64_t fixedOne = (1LL << fixedShift);
+    static const int64_t fixedRound = ((fixedOne >> 1) - 1);
+
+    int64_t val;
+    fixedpoint64(int64_t _val) : val(_val) {}
+    static CV_ALWAYS_INLINE uint64_t fixedround(const uint64_t& _val) { return (_val + fixedRound + ((_val>>fixedShift)&1)); } // banker's rounding
```

Comment:
`fixedround` is already defined as static const.
Static code analyzers dislikes this.